### DISCUSSION TITLE
Add pipe option to read_csv_auto

### DISCUSSION THREAD

**Discussion by mmoya** [duckdb/duckdb#6691] [Ideas]
`sqlite3` allows to import a file with the `.import` command but [if the filename starts with a `|`, it specifies a command which will be run to produce the input data](https://sqlite.org/cli.html#importing_files_as_csv_or_other_formats).

A typical use case is to directly query a CSV without having to wrap the execution in a script that stores the output in a temporary file.

For example, the following command will be a very succint way to express _please give me an immediate query prompt on the CSV output generated by this other command_:

```
$ duckdb -cmd "CREATE TABLE data FROM read_csv_auto('|generate-csv-output')"
v0.7.1 b00b93f0b1
Enter ".help" for usage hints.
D SELECT COUNT(*) FROM data;
count_star()
42
D 
```

The following command is not an option because all columns are parsed as `VARCHAR`:

```
duckdb -cmd '.mode csv' -cmd ".import '|generate-csv-output' data"
```

**Comment by lnkuiper** [comment]
Does this work for you?
```sql
$ echo '1,2,3' | duckdb -c "select * from read_csv_auto('/dev/stdin');"
┌─────────┬─────────┬─────────┐
│ column0 │ column1 │ column2 │
│  int64  │  int64  │  int64  │
├─────────┼─────────┼─────────┤
│       1 │       2 │       3 │
└─────────┴─────────┴─────────┘
```

**Reply by lnkuiper** [reply]
DuckDB can also read unix pipes made by, e.g., `mkfifo`

**Reply by mmoya** [reply]
Using `/dev/stdin` works but you won't be given a prompt after reading the CSV.

 The usecase is:

1. I want a prompt to interactively play with the data, I don't know queries beforehand
2. The csv is generated by a command and written to its stdout

Rationale behind the feature request:

- Reading from a pipe follows the unix tradition of composing commands with pipes. This is not exactly `a | b` but is near.
- Queries are not know beforehand. Duckdb here is used precisely to play with the CSV using SQL.
- The `|command` syntax is already supported by duckdb and sqlite `.import`.
- It'd be like the marriage of `.import` and `read_csv_auto`.

the workaround now is to use a temporary file and a pre command:

```sh
tmpfile=$(mktemp)
generate-csv-output >$tmpfile
duckdb -cmd "CREATE TABLE data AS SELECT * FROM read_csv_auto('$tmpfile')"
```

Actually I don't know if its practical or desirable to have a way to exhaust the stdin before dropping to a prompt. Thinking in something like:

```
$ alias querycsv="duckdb -cmd 'CREATE TABLE data AS SELECT * FROM read_csv_auto_and_exhaust_stdin(\'/dev/stdin\')"

$ querycsv <existing-file.csv
D 

$ writes-csv-to-stdout | querycsv
D
```

this would be number 1 answer to 'how to query CSV data with SQL in the command line'.
